ddtrace/internal/remoteconfig/_subscribers.py

```diff
@@ -0,0 +1,110 @@
+import abc
+import os
+import time
+
+import six
+
+from ddtrace.internal.logger import get_logger
+from ddtrace.internal.periodic import PeriodicThread
+from ddtrace.internal.remoteconfig.utils import get_poll_interval_seconds
+
+
+log = get_logger(__name__)
+
+
+class SubscriberBase(six.with_metaclass(abc.ABCMeta)):
+    def __init__(self, data_connector, callback, name):
+        self._data_connector = data_connector
+        self.is_running = False
+        self._callback = callback
+        self._name = name
+        log.debug("[%s] Subscriber %s init", os.getpid(), self._name)
+
+    @abc.abstractmethod
+    def start(self):
+        pass
+
+    @abc.abstractmethod
+    def stop(self):
+        pass
+
+    @abc.abstractmethod
+    def force_restart(self):
+        pass
+
+
+class SubscriberMock(SubscriberBase):
+    def start(self):
+        pass
+
+    def stop(self):
+        pass
+
+    def force_restart(self):
+        pass
+
+
+class RemoteConfigSubscriber(SubscriberBase):
+    _th_worker = None
+
+    def __init__(self, data_connector, callback, name):
+        super(RemoteConfigSubscriber, self).__init__(data_connector, callback, name)
+
+    def _exec_callback(self, data, test_tracer=None):
+        if data:
+            log.debug("[%s] Subscriber %s _exec_callback", os.getpid(), self._name)
+            self._callback(data, test_tracer=test_tracer)
+
+    def _get_data_from_connector_and_exec(self, checksum=0, test_tracer=None):
+        data = self._data_connector.read()
+        data_raw = str(data)
+        last_checksum = hash(data_raw)
```

Comment:
Good Idea! I'm going to update with your idea :+1:

Reply:
I'm reviewing the code and I think that we can do this check on the connector classes because this classes are the "pipe" between publisher and subscriber and  I think they are the responsible to control if the information should pass between both of them, what do you think @P403n1x87 ?